### ISSUE TITLE
Replace Heroku with something else (GitHub Actions?)

### ISSUE THREAD

**Issue by pllim** [astropy/astropy-bot#125]
Reasons:

* Heroku + GitHub API been flaky lately.
* Heroku logs hard to decipher.
* Heroku access is a bottleneck and extra maintenance burden.
* Updating upstream dependency on Heroku can be tricky because it is cached.

This issue supersedes #122 .

## Replacements that are already deployed in astropy

* `astropy_bot/autolabel.py` ➡️ [labeler](https://github.com/actions/labeler) (see `astropy`)
* `baldrick/plugins/github_pull_requests_base_branch.py` ➡️ astropy/astropy#11158 using [actions/github-script](https://github.com/actions/github-script) (POCs [here](https://github.com/kali-maa/astropy/pull/43), [here](https://github.com/kali-maa/astropy/pull/65), and [here](https://github.com/kali-maa/astropy/pull/66))
* `baldrick/plugins/github_towncrier_changelog.py` ➡️ [actions-towncrier-changelog](https://github.com/pllim/actions-towncrier-changelog) (astropy/astropy#11489 is a stop-gap fix until @Cadair can figure something out as permanent fix)
* `baldrick/scripts/stale_issues.py` and `baldrick/scripts/stale_pull_requests.py` ➡️ astropy/astropy#12131

## Replacements that work but not deployed

* `baldrick/utils.py` ➡️ [action-special_pr_comment](https://github.com/pllim/action-special_pr_comment) (POCs [here](https://github.com/kali-maa/astropy/issues/48), [here](https://github.com/kali-maa/astropy/issues/51), [here](https://github.com/kali-maa/astropy/pull/49), [here](https://github.com/kali-maa/astropy/pull/50), and [here](https://github.com/kali-maa/astropy/pull/52))
  * p.s. I did temporarily deploy it in `astropy` on "special day" when `astropy-bot` was broken but I commented it back out after.

## Replacements that need upstream fixes

* `baldrick/plugins/github_milestones.py` ➡️ See the `github-script` example in [action-check_milestone_exists](https://github.com/pllim/action-check_milestone_exists) (blocked by [feature-request-add-milestone-changes-as-activity-type-to-pull-request](https://github.community/t/feature-request-add-milestone-changes-as-activity-type-to-pull-request/16778/7))

## No longer relevant

* `astropy_bot/changelog_checker.py` ➡️ [action-check_astropy_changelog](https://github.com/pllim/action-check_astropy_changelog) (POCs [here](https://github.com/kali-maa/astropy/pull/28) and [here](https://github.com/kali-maa/astropy/pull/29)) -- This has been made obsolete by astropy/astropy#10334

**Comment by pllim** [comment]
I took a stab at replacing the milestone check with [custom GitHub Action](https://github.com/pllim/action-check_milestone_exists) but it is blocked by https://github.community/t/feature-request-add-milestone-changes-as-activity-type-to-pull-request/16778/7 .

From that exercise, my conclusions are as follow: It will be pretty much a complete refactoring moving from astropy-bot to Actions. A lot of the things that make sense now won't make sense in Actions. For example, we don't need TOML config because you can customize the Actions itself. We can use PyGithub instead of custom GitHub API, probably no need for Flask app too, etc etc.

Also not sure how to write unit tests for a custom GitHub Action.

UPDATE: Going the GitHub Action route, implementing it in TypeScript would be much more efficient. Using Python + Docker has at least a 20s overhead and that is using a custom Docker image stored at quay.io with dependencies installed.

**Comment by pllim** [comment]
Priorities...

![Screenshot 2020-11-21 203607](https://user-images.githubusercontent.com/2090236/99891565-3b06ee00-2c39-11eb-8601-f85603adf58a.jpg)
